### PR TITLE
HPCC-13515 Updates to readme to provide a more general overview, and links to details.

### PR DESCRIPTION
**Pull request by boneill42**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,102 +1,63 @@
...
+* Single-threaded
+* Distributed parallel processing
+* Distributed file system
+* Powerful parallel processing programming language (ECL)
+* Optimized for Extraction, Transformation, Loading, Sorting, Indexing and Linking
+* Scales from 1-1000s of nodes
+
+## [Roxie](http://hpccsystems.com/FAQ/what-roxie)
+
+Roxie (the Query Cluster) provides separate high-performance online query processing and data warehouse capabilities.  Roxie (Rapid Online XML Inquiry Engine) is the data delivery engine used in HPCC to serve data quickly and can support many thousands of requests per node per second.
```
Comment (richardkchapman) [inline README.md]: Roxie stands for Richard's Online XML Inquiry Engine (despite what some of our literature may claim)

File: README.md [hunk 2]
```diff
@@ -1,102 +1,63 @@
...
+
+## [ECL](http://hpccsystems.com/FAQ/what-is-ecl)
+
+ECL (Enterprise Control Language) is the powerful programming language that is ideally suited for the manipulation of Big Data.
+
+* Transparent and implicitly parallel programming language
+* Non-procedural and dataflow oriented
+* Modular, reusable, extensible syntax
+* Combines data representation and algorithm implementation
+* Easily extend using C++ libraries
```
Comment (richardkchapman) [inline README.md]: Worth mentioning that you can embed Python, JavaScript, etc

File: README.md [hunk 3]
```diff
@@ -1,102 +1,63 @@
...
+
+# Architecture
+
+The HPCC Systems architecture incorporates the Thor and Roxie clusters as well as common middleware components, an external communications layer, client interfaces which provide both end-user services and system management tools, and auxiliary components to support monitoring and to facilitate loading and storing of filesystem data from external sources. An HPCC environment can include only Thor clusters, or both Thor and Roxie clusters. Each of these cluster types is described in more detail in the following sections below the architecture diagram.
+
+## [Thor](http://hpccsystems.com/FAQ/what-thor)
+
+Thor (the Data Refinery Cluster) is responsible for consuming vast amounts of data, transforming, linking and indexing that data. It functions as a distributed file system with parallel processing power spread across the nodes. A cluster can scale from a single node to thousands of nodes.
+
+* Single-threaded
```
Comment (mckellyln) [inline README.md]: Thor is multi-threaded

Comment (richardkchapman) [inline README.md]: I think the distinction Brian is trying to make is that thor runs one query at a time while Roxie runs multiple. Internally, both use multiple threads (and multiple processes) to get the job done.

Comment (mckellyln) [inline README.md]: ok, I concur.  But perhaps threaded is not the best word to use here (but I cannot think of anything better at the moment).

Comment (richardkchapman) [inline README.md]: Multiple concurrent queries / Single job runs at a time

**Comment by boneill42** [comment]
Sorry -- I can't take any credit here...

I actually just copied content from this page:
http://hpccsystems.com/Why-HPCC/How-it-works

File: README.md (hunk 2 above)
Comment (boneill42) [inline README.md]: Agreed, will update.

**Comment by richardkchapman** [comment]
@boneill42 Looks like this is waiting for a couple of minor updates from you before I merge it.

**Comment by AttilaVamos** [comment]
Automated Smoketest Build: success

**Comment by boneill42** [comment]
@richardkchapman Comments should be address with the latest commit.

**Comment by richardkchapman** [comment]
You should not merge latest master into your branch - rather, you should rebase your branch so that it is relative to latest master. That way your branch will only have the changes you have made rather than a second copy of all the other changes (which would get very confusing).

**Comment by richardkchapman** [comment]
Cleaned up and merged manually to master

**Pull request closed without merging**